perception/sub8_perception/nodes/path_marker_finder.py

```diff
@@ -205,22 +214,15 @@ def valid_contour(self, contour):
         '''
         if cv2.contourArea(contour) < self.min_contour_area:
             return False
+        match_shapes = cv2.matchShapes(contour, PathMarkerFinder.PATH_MARKER_2D, 3, 0.0)
+        if match_shapes > self.shape_match_thresh :
+            return False
         # Checks that contour is 4 sided
         polygon = cv2.approxPolyDP(contour, self.approx_polygon_thresh, True)
         if len(polygon) != 4:
+            #rospy.loginfo("Polygon not 4 sided ({}), throwing out".format(len(polygon)))
```

Comment:
This usually just indicates a contour that isnt the park marker, so it would just spam dozens of non rectangle contours each frame